### PR TITLE
Maintenance/use pytest simcore for system tests/director integration tests

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

**pytest-simcore**:
- docker-registry small fixes
- docker-compose was not using locally built images but production images which leads to inconsistencies in the tests
- docker-swarm now checks for the IP (useful if more than one network interface is in use)
- minio-service provides bucket fixture + fix S3_SECURE env
- postgres-service follows same schema as other fixtures using a tenacitied function
- rabbit-service: fixed unclosed connection. adds rabbit connection, channels,queues,exchanges
- storage fixture: fix minor issue

**system-testing**:
- uses pytest-simcore
- fix syntax differences between alpine/debian based

**director**:
- uses python-3.6.0 dockerfile based on Debian
- uses pytest-simcore

## Checklist

- [ ] Did you change any service's API? Then make sure to bundle document and upgrade version (``make openapi-specs``, ``git commit ...`` and then ``make version-*``)
- [ ] Unit tests for the changes exist
- [ ] Runs in the swarm
- [ ] Documentation reflects the changes
- [ ] New module? Add your github username to [.github/CODEOWNERS](.github/CODEOWNERS)

### CODE REVIEW

File: packages/pytest-simcore/src/pytest_simcore/docker_compose.py [hunk 1]
```diff
@@ -77,9 +77,20 @@ def env_file(osparc_simcore_root_dir: Path, devel_environ: Dict[str, str]) -> Pa
         backup_path.unlink()
 
 
+@pytest.fixture(scope="module")
+def make_up_prod_environ():
+    if not "DOCKER_REGISTRY" in os.environ:
+        os.environ["DOCKER_REGISTRY"] = "local"
+    if not "DOCKER_IMAGE_TAG" in os.environ:
+        os.environ["DOCKER_IMAGE_TAG"] = "production"
+
```
Comment (pcrespov) [inline packages/pytest-simcore/src/pytest_simcore/docker_compose.py]: the variables need to be unset when the test fixture gets out of scope. My proposal:
```python
@pytest.fixture(scope="module")
def make_up_prod_environ(monkeypatch):
    if not "DOCKER_REGISTRY" in os.environ:
         monkeypatch.setenv("DOCKER_REGISTRY", "local")
  ...
```
here monkeypatch takes care of unsetting them once the fixture is done

More info and examples in https://docs.pytest.org/en/3.0.1/monkeypatch.html#example-setting-an-environment-variable-for-the-test-session

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 2]
```diff
@@ -20,21 +21,24 @@ def docker_client() -> docker.client.DockerClient:
     yield client
 
 
+@pytest.fixture(scope="session")
+def keep_docker_up(request) -> bool:
+    return request.config.getoption("--keep-docker-up") == True
```
Comment (pcrespov) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py]: MINOR: no need to ``==True`` here

File: packages/pytest-simcore/src/pytest_simcore/docker_swarm.py [hunk 3]
```diff
@@ -20,21 +21,24 @@ def docker_client() -> docker.client.DockerClient:
...
+def docker_swarm(
+    docker_client: docker.client.DockerClient, keep_docker_up: bool
+) -> None:
     try:
         docker_client.swarm.reload()
         print("CAUTION: Already part of a swarm")
         yield
     except docker.errors.APIError:
-        docker_client.swarm.init()
+        docker_client.swarm.init(advertise_addr=get_ip())
```
Comment (pcrespov) [inline packages/pytest-simcore/src/pytest_simcore/docker_swarm.py]: interesting ... what is the purpose of advertising?

**Comment by sanderegg** [comment]
no viable

**Pull request closed without merging**